src/com/widen/util/S3UrlBuilder.java

```diff
@@ -330,7 +332,11 @@ else if (BucketEncoding.VIRTUAL_DNS.equals(requestedBucketEncoding))
 		{
 			canSign();
 
-			builder.addParameter("response-content-disposition", String.format("attachment; filename=\"%s\"", attachmentFilename));
+			// Most browsers don't support UTF-8 in HTTP headers (HTTP officially supports only ISO-8859-1). In
+			// practice, S3 only supports ASCII characters, so strip everything out not in the ASCII range.
+			String strippedFilename = attachmentFilename.replaceAll("[^\\x00-\\x7F]", "");
```

Comment:
Running the name though Normalizer would help preserve some characters... http://stackoverflow.com/a/17786019/96224

Regex could be tightened to strip control characters: `[^\\x20-\\x7E]`